frappe/database/database.py

```diff
@@ -677,53 +692,55 @@ def set_value(self, dt, dn, field, val=None, modified=None, modified_by=None,
 		:param debug: Print the query in the developer / js console.
 		:param for_update: Will add a row-level lock to the value that is being set so that it can be released on commit.
 		"""
-		if not modified:
-			modified = now()
-		if not modified_by:
-			modified_by = frappe.session.user
+		is_single_doctype = not (dn and dt != dn)
+		to_update = field if isinstance(field, dict) else {field: val}
 
-		to_update = {}
 		if update_modified:
-			to_update = {"modified": modified, "modified_by": modified_by}
+			modified = modified or now()
+			modified_by = modified_by or frappe.session.user
+			to_update.update({"modified": modified, "modified_by": modified_by})
+
+		if is_single_doctype:
+			frappe.db.delete(
+				"Singles",
+				filters={"field": ("in", tuple(to_update)), "doctype": dt}, debug=debug
+			)
+
+			singles_data = ((dt, key, sbool(value)) for key, value in to_update.items())
```

Comment:
Any specific reason for `sbool`? `sbool` isn't JUST converting truthy/falsy values it will also assume more:

<img width="1265" alt="Screenshot 2022-01-17 at 12 48 14 PM" src="https://user-images.githubusercontent.com/9079960/149724528-a86ae303-082f-4e92-9575-5fd763c0a8e5.png">

There are some singles where brilliant minds have used "yes"/"no" instead of checkbox which could break after this change 😄

Reply:
Fixed (in sbool), only converting values like "true" and "1" now to bool